reVX/plexos/rev_reeds_plexos.py

```diff
@@ -1,1460 +1,1134 @@
+                    bad_sc_points.append(self._sc_build.iloc[i]['gid'])
+
+            wmsg = ('There are {} SC points with missing gids: {}'
+                    .format(len(bad_sc_points), bad_sc_points))
+            warn(wmsg)
+            logger.warning(wmsg)
+
+        return bad_sc_points
+
+    def _handle_missing_resource_gids(self, bad_sc_points):
+        """Merge requested capacity in missing SC gids into nearest good pixels
+
+        Parameters
+        ----------
+        bad_sc_points : list
+            List of missing supply curve gids
+            (in reeds but not in reV resource).
+        """
+        if any(bad_sc_points):
+            bad_bool = self._sc_build['gid'].isin(bad_sc_points)
+            bad_cap_arr = self._sc_build.loc[bad_bool, 'built_capacity'].values
+            good_bool = ~bad_bool
+            bad_cap = bad_cap_arr.sum()
+            wmsg = ('{} MW of capacity is being merged from bad SC points.'
+                    .format(bad_cap))
+            warn(wmsg)
+            logger.warning(wmsg)
+
+            clabels = get_coord_labels(self._sc_build)
+            good_tree = cKDTree(self._sc_build.loc[good_bool, clabels])
+            _, i = good_tree.query(self._sc_build.loc[bad_bool, clabels])
+
+            ilen = len(self._sc_build)
+            icap = self._sc_build['built_capacity'].sum()
+
+            add_index = self._sc_build.index.values[good_bool][i]
+
+            for i, ai in enumerate(add_index):
+                self._sc_build.loc[ai, 'built_capacity'] += bad_cap_arr[i]
+
+            bad_ind = self._sc_build.index.values[bad_bool]
+            self._sc_build = self._sc_build.drop(bad_ind, axis=0)
+
+            olen = len(self._sc_build)
+            ocap = self._sc_build['built_capacity'].sum()
+
+            wmsg = ('SC build table reduced from {} to {} rows, '
+                    'capacity from {} to {} (should be the same).'
+                    .format(ilen, olen, icap, ocap))
+            warn(wmsg)
+            logger.warning(wmsg)
+
+            cap_error = (icap - ocap) / icap
+            if cap_error > 0.001:
+                msg = ('Too much capacity is being lost due to missing '
+                       'resource gids! Capacity difference is {}%. '
+                       'Cannot continue.'.format(cap_error * 100))
+                logger.error(msg)
+                raise RuntimeError(msg)
+
+    def _init_output(self):
+        """Init the output array of aggregated PLEXOS profiles.
+
+        Returns
+        -------
+        output : np.ndarray
+            (t, n) array of zeros where t is the timeseries length and n is
+            the number of plexos nodes.
+        """
+
+        if self._forecast_fpath is None:
+            with Outputs(self._cf_fpath, mode='r') as out:
+                t = len(out.time_index)
+        else:
+            with Outputs(self._forecast_fpath, mode='r') as out:
+                t = len(out.time_index)

```

Comment:
can you not use out.shape[0]? This would be a lot faster

Reply:
agreed